kafka-streams/src/main/java/io/confluent/developer/errors/StreamsErrorHandling.java

```diff
@@ -55,7 +58,10 @@ public ProductionExceptionHandlerResponse handle(ProducerRecord<byte[], byte[]>
                                                          Exception exception) {
             // This return null statement is here so the code will compile
             // You need to replace it with some logic described below
-            return null;
+            if(exception instanceof RecordTooLargeException){
+                return ProductionExceptionHandlerResponse.CONTINUE;
+            }
+            return ProductionExceptionHandlerResponse.FAIL;
             // If the exception type is RecordTooLargeException continue working
```

Comment:
Same as above - there are a bunch of other changes below that follow the same pattern, but I won't repeat the comments.